corehq/apps/userreports/tasks.py

```diff
@@ -303,119 +304,139 @@ def _queue_chunk(indicators):
-        indicator_config_ids = first_indicator.indicator_config_ids
-        related_docs_to_rebuild = set()
+        doc_store = get_document_store_for_doc_type(
+            all_indicators[0].domain, all_indicators[0].doc_type
+        )
+        related_doc_ids = set()
+        failed_indicators = set()
 
+        rows_to_save_by_adapter = defaultdict(list)
+        doc_ids_to_save_by_adapter = defaultdict(list)
+        indicator_by_doc_id = {i.doc_id: i for i in all_indicators}
+        config_ids = set()
         with timer:
             for doc in doc_store.iter_documents(list(indicator_by_doc_id.keys())):
                 indicator = indicator_by_doc_id[doc['_id']]
-                successfully_processed, to_remove, rebuild_related_docs = _save_document_helper(indicator, doc)
-                if rebuild_related_docs:
-                    related_docs_to_rebuild = related_docs_to_rebuild.union(icds_get_related_docs_ids(doc['_id']))
-                if successfully_processed:
-                    processed_indicators.append(indicator.pk)
+                eval_context = EvaluationContext(doc)
+                for config_id in indicator.indicator_config_ids:
+                    config_ids.add(config_id)
+                    try:
+                        config = _get_config(config_id)
+                    except (ResourceNotFound, StaticDataSourceConfigurationNotFoundError):
+                        celery_task_logger.info("{} no longer exists, skipping".format(config_id))
+                        # remove because the config no longer exists
+                        _mark_config_to_remove(config_id, [indicator.pk])
+                        continue
+                    except ESError:
+                        celery_task_logger.info("ES errored when trying to retrieve config")
+                        failed_indicators.add(indicator)
+                        continue
+                    adapter = None
+                    try:
+                        adapter = get_indicator_adapter(config, can_handle_laboratory=True)
+                        rows_to_save_by_adapter[adapter].extend(adapter.get_all_values(doc, eval_context))
+                        doc_ids_to_save_by_adapter[adapter].append(doc['_id'])
+                        eval_context.reset_iteration()
+                    except Exception as e:
+                        failed_indicators.add(indicator)
+                        handle_exception(e, config_id, doc, adapter)
+
+                    if config and config.icds_rebuild_related_docs:
+                        related_doc_ids.add(doc['_id'])
+
+            for adapter, rows in six.iteritems(rows_to_save_by_adapter):
+                doc_ids = doc_ids_to_save_by_adapter[adapter]
+                indicators = [indicator_by_doc_id[doc_id] for doc_id in doc_ids]
+                try:
+                    adapter._save_rows(rows, doc_ids)
```

Comment:
This is accessing a hidden function of the adapter. Should be changed to `save_rows`

Reply:
Sure